source/pool/contracts/Pool.sol

```diff
@@ -257,13 +276,21 @@ contract Pool is IPool, Ownable2Step {
     Claim memory _claim;
     for (uint256 i = 0; i < _claims.length; i++) {
```

Comment:
Store the root in memory for the duration of each loop to avoid 4 or 5 extra reads from storage and save gas.

`bytes32 root = rootsByGroupId[_claim.groupId];`